### PR TITLE
Refactor more dependency notations to the new DependencyGroup API

### PR DESCRIPTION
**Pull request by jmfayard**

### CODE REVIEW

File: plugins/dependencies/src/test/kotlin/de/fayard/NonRegression.kt [hunk 1]
```diff
@@ -16,6 +18,17 @@ val testResources: File = File(".").absoluteFile.resolve("src/test/resources")
...
+        ALL_DEPENDENCIES_NOTATIONS
+        val rulesDir = File(".").absoluteFile.resolve("src/main/resources/refreshVersions-rules")
+        val file = rulesDir.resolve("dependency-groups-alias-rules.txt")
+        val content = AbstractDependencyGroup.ALL_RULES
+            .sorted()
+            .distinct()
+            .joinToString(separator = "\n\n")
+        file.writeText(content)
+    }
+
```
Comment (LouisCAD) [inline plugins/dependencies/src/test/kotlin/de/fayard/NonRegression.kt:31]: @jmfayard Why did you add that?

Comment (LouisCAD) [reply plugins/dependencies/src/test/kotlin/de/fayard/NonRegression.kt:31]: Oh wait, I see that you actually moved it from one file to another. Still wondering why though.

Comment (LouisCAD) [reply plugins/dependencies/src/test/kotlin/de/fayard/NonRegression.kt:31]: I mean, this test is not about non regression. Maybe we should rename this file to something like `BundledDependencyNotationsTests`?

Comment (jmfayard) [reply plugins/dependencies/src/test/kotlin/de/fayard/NonRegression.kt:31]: @LouisCAD I only moved it from another test because it belongs together
